### ISSUE TITLE
predict(newdata =,)) issue when using multiple transformations of same variable 

### ISSUE THREAD

**Issue by willtudorevans** [glmmTMB/glmmTMB#853]
I seem to be getting very strange predictions when using predict(newdata =)) with a glmmTMB mode. Predictions that don't align with fitted(mod) values if I rebuild the predictions on trained data. The issue seems to arise if I use a second transformation of the same variable, and particularly if that transformation involves a '+' sign.

Apologies for the esoteric reproducible example but I hope it demonstrates the point ok. I'm using glmmTMB v1.1.4.

```
rm(list = ls())
library(glmmTMB)
set.seed(100)
N <- 100
x1 <- rnorm(N)
Y <- rpois(N, lambda = exp(x1))
df <- data.frame(Y=Y, x1=x1)

#Base model
mod <- glmmTMB(Y ~ x1 + I((x1+10)^2) + I((x1+10)^3), data = df, family = "poisson")
all(round(fitted(mod),2) == round(predict(mod, newdata = df, type = "response"),2))
#not ok

#Rearrange ordering of second transformation
mod <- glmmTMB(Y ~ x1 + I((x1+10)^2) + I((10+x1)^3), data = df, family = "poisson")
all(round(fitted(mod),2) == round(predict(mod, newdata = df, type = "response"),2))
#ok

#Remove second transformation
mod <- glmmTMB(Y ~ x1 + I((x1+10)^2), data = df, family = "poisson")
all(round(fitted(mod),2) == round(predict(mod, newdata = df, type = "response"),2))
#ok

#Try glm of base model
mod <- glm(Y ~ x1 + I((x1+10)^2) + I((x1+10)^3), data = df, family = "poisson")
all(round(fitted(mod),2) == round(predict(mod, newdata = df, type = "response"),2))
#ok

#Add fourth transformation
mod <- glmmTMB(Y ~ x1 + I((x1+10)^2) + I((10+x1)^3) + I((10+x1)^4), data = df, family = "poisson")
all(round(fitted(mod),2) == round(predict(mod, newdata = df, type = "response"),2))
#not ok

#Add fourth transformation with different starting point
mod <- glmmTMB(Y ~ x1 + I((x1+10)^2) + I((10+x1)^3) + I((5+x1)^4), data = df, family = "poisson")
all(round(fitted(mod),2) == round(predict(mod, newdata = df, type = "response"),2))
#ok

#Remove additions from transformations
mod <- glmmTMB(Y ~ x1 + I(x1^2) + I(x1^3), data = df, family = "poisson")
all(round(fitted(mod),2) == round(predict(mod, newdata = df, type = "response"),2))
#ok
```

**Comment by bbolker** [comment]
This is definitely the same problem as #845 (and #70, and [this comment](https://github.com/glmmTMB/glmmTMB/blob/e1f32b8fa3d4ba6626e58190eef71c59c71d8d02/glmmTMB/R/predict.R#L241-L252) that says we know that `fix_predvars` is ugly ...

**Comment by bbolker** [comment]
Can you please `remotes::install_github("glmmTMB/glmmTMB/glmmTMB@more_predvars")` and check if this resolves the issue?  Further crazy test cases welcome ...

**Comment by bbolker** [comment]
@willtudorevans  Can you comment on whether this is fixed in https://github.com/glmmTMB/glmmTMB/pull/854 ?

**Comment by bbolker** [comment]
Presumably closed in #854.

**Comment by willtudorevans** [comment]
Sorry for the slow response. Yes this update has fixed the issue, thank you very much.
